ui/components/app/confirmations/alerts/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Text,
+} from '../../../../component-library';
+import {
+  Display,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import useAlerts from '../../../../../hooks/useAlerts';
+import { AlertModal } from '../alert-modal';
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+
+export type MultipleAlertModalProps = {
+  /** The key of the initial alert to display. */
+  alertKey: string;
+  /** The function to be executed when the modal needs to be closed. */
+  onClose: () => void;
+  /** The function to be executed when the button in the alert modal is clicked. */
+  onFinalAcknowledgeClick: () => void;
+  /** The unique identifier of the entity that owns the alert. */
+  ownerId: string;
+};
+
+function PageNavigation({
+  alerts,
+  onBackButtonClick,
+  onNextButtonClick,
+  selectedIndex,
+}: {
+  alerts: Alert[];
+  onBackButtonClick: () => void;
+  onNextButtonClick: () => void;
+  selectedIndex: number;
+}) {
+  const t = useI18nContext();
+  const showPreviousButton = selectedIndex + 1 > 1;
+  const showNextButton = selectedIndex + 1 < alerts.length;
+
+  function PreviousButton() {
+    if (!showPreviousButton) {
+      return null;
+    }
+
+    return (
+      <ButtonIcon
+        iconName={IconName.ArrowLeft}
+        ariaLabel={t('back')}
+        size={ButtonIconSize.Sm}
+        onClick={onBackButtonClick}
+        className={'multiple-alert-modal__arrow-buttons'}
+        data-testid="alert-modal-back-button"
+      />
+    );
+  }
+
+  function NextButton() {
+    if (!showNextButton) {
+      return null;
+    }
+
+    return (
+      <ButtonIcon
+        iconName={IconName.ArrowRight}
+        ariaLabel={t('next')}
+        size={ButtonIconSize.Sm}
+        onClick={onNextButtonClick}
+        className={'multiple-alert-modal__arrow-buttons'}
+        data-testid="alert-modal-next-button"
+      />
+    );
+  }
+
+  function PageNumber() {
+    return (
+      <Text
+        variant={TextVariant.bodySm}
+        color={TextColor.textAlternative}
+        className={'multiple-alert-modal__text'}
+      >
+        {`${selectedIndex + 1} ${t('ofTextNofM')} ${alerts.length}`}
+      </Text>
+    );
+  }
+
+  return (
+    <Box display={Display.Flex}>
+      <PreviousButton />
+      <PageNumber />
+      <NextButton />
+    </Box>
+  );
+}
+
+export function MultipleAlertModal({
+  alertKey,
+  onClose,
+  onFinalAcknowledgeClick,
+  ownerId,
+}: MultipleAlertModalProps) {
+  const { alerts } = useAlerts(ownerId);
+
+  const [selectedIndex, setSelectedIndex] = useState(
+    alerts.findIndex((alert) => alert.key === alertKey),
+  );
+
+  const selectedAlert = useMemo(
```

Comment:
If this is dependent on the alerts and the selected index, then is a memo needed here?

Isn't this equivalent to just extracting on each render?

Reply:
yes. I removed memo from this line.